cmd/runtimetest/main.go

```diff
@@ -320,6 +322,95 @@ func validateROPaths(spec *rspec.Spec) error {
 	return nil
 }
 
+func getIDMappings(idType string) ([]rspec.IDMapping, error) {
+	var mapFile string
+	var idMaps []rspec.IDMapping
+	switch idType {
+	case "uid_map":
+		mapFile = "/proc/1/uid_map"
+	case "gid_map":
+		mapFile = "/proc/1/gid_map"
+	}
+	f, err := os.Open(mapFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return nil, err
+		}
+
+		text := strings.Replace(strings.TrimSpace(s.Text()), "          ", " ", -1)
+		idMap := strings.Split(text, " ")
+		if len(idMap) == 3 {
+			hostID, err := strconv.ParseUint(idMap[0], 0, 32)
+			if err != nil {
+				return nil, err
+			}
+			containerID, err := strconv.ParseUint(idMap[1], 0, 32)
+			if err != nil {
+				return nil, err
+			}
+			mapSize, err := strconv.ParseUint(idMap[2], 0, 32)
+			if err != nil {
+				return nil, err
+			}
+			idMaps = append(idMaps, rspec.IDMapping{HostID: uint32(hostID), ContainerID: uint32(containerID), Size: uint32(mapSize)})
+		} else {
+			return nil, fmt.Errorf("invalid format in %v", mapFile)
+		}
+	}
+
+	return idMaps, nil
+}
+
+func validateUIDMappings(spec *rspec.Spec) error {
+	logrus.Debugf("validating uidMappings")
+	idMaps, err := getIDMappings("uid_map")
+	if err != nil {
+		return fmt.Errorf("can not get items from uid_map: %v", err)
+	}
+	for _, v := range spec.Linux.UIDMappings {
+		exist := false
+		for _, cv := range idMaps {
+			if v.HostID == cv.HostID && v.ContainerID == cv.ContainerID && v.Size == cv.Size {
+				exist = true
+				break
+			}
+		}
+		if !exist {
+			return fmt.Errorf("uidMappings is not applied as expected")
+		}
+	}
+
+	return nil
+}
+
+func validateGIDMappings(spec *rspec.Spec) error {
```

Comment:
This is very similar to `validateUIDMappings`  Can we pull out the common code into:

    validateIDMappings(mappings []IDMapping, path string, property string) error

and then use:

    func validateUIDMappings(spec *rspec.Spec) error {
        return validateIDMappings(spec.Linux.UIDMappings, "/proc/self/uid_map", "linux.uidMappings")
    }

    func validateGIDMappings(spec *rspec.Spec) error {
        return validateIDMappings(spec.Linux.GIDMappings, "/proc/self/gid_map", "linux.gidMappings")
    }